src/Microsoft.Health.Dicom.Core/Features/Telemetry/InstanceMeter.cs

```diff
@@ -0,0 +1,30 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Health.Dicom.Core.Features.Telemetry;
+public sealed class InstanceMeter : IDisposable
+{
+    private readonly Meter _meter;
+
+    public InstanceMeter()
+    {
+        _meter = new Meter("Microsoft.Health.Dicom.Core.Features.Telemetry.Instance", "1.0");
+        InstanceCount = _meter.CreateCounter<double>(nameof(InstanceCount));
+        TotalInstanceBytes = _meter.CreateHistogram<double>(nameof(TotalInstanceBytes));
+        MinInstanceBytes = _meter.CreateHistogram<double>(nameof(MinInstanceBytes));
+        MaxInstanceBytes = _meter.CreateHistogram<double>(nameof(MaxInstanceBytes));
+    }
+
+    public Counter<double> InstanceCount { get; }
+    public Histogram<double> TotalInstanceBytes { get; }
+    public Histogram<double> MinInstanceBytes { get; }
```

Comment:
Do we need separate `Histogram<T>` metrics for min, max, and total? Doesn't the histogram encapsulate them all? Even the count? https://opentelemetry.io/docs/reference/specification/metrics/data-model/#histogram

Reply:
I don't think so. To my understanding, histogram cannot be used to track multiple values at the same time

Reply:
Are you sure about this? I thought the whole point of a histogram was to be a set of data points. See here about historgrams in Azure Monitor: https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=net#add-custom-metrics

Historgrams automatically seem to support the following aggregation types: Min, Max, Average, Sum and Count

Reply:
Absolutely right, it is a set of data points. But in our case, the aggregation is already done, and we have the min, max and sum as separate metrics.

Reply:
Synced offline and grouped all the metrics into a single histogram `InstanceLength`